src/main/java/com/google/codeu/servlets/ChartsDataServlet.java

```diff
@@ -72,11 +73,51 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 		else if(analysisType.equals("lastSevenDays")) {
 			this.getLastSevenDays(request, response);
 		}
+		else if(analysisType.equals("breakdown")) {
+			this.getBreakdown(request, response);
+		}
 		else {
 			response.sendError(400, "Invalid analysis type");
 		}
 	}
 	
+	/*
+	 * Returns JSON response of UserMeals by category: breakfast, lunch, dinner, and snack
+	 * Example response: { "breakfast": [meal1, meal2], "lunch": [meal1, meal2] };
+	 * */
+	private void getBreakdown(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		response.setContentType("application/json");
+		String user = request.getParameter("user");
+		
+		JsonObject json = new JsonObject();
+		json.addProperty("breakfast", this.categoryQuery(user, "breakfast"));
+		json.addProperty("lunch", this.categoryQuery(user, "lunch"));
+		json.addProperty("dinner", this.categoryQuery(user, "dinner"));
+		json.addProperty("snack", this.categoryQuery(user, "snack"));
+		
```

Comment:
not sure if you could change the string to integer values since Debbie used Enum.ordinal()